samtranslator/model/apigateway.py

```diff
@@ -95,6 +95,9 @@ def make_auto_deployable(self, stage, openapi_version=None, swagger=None):
         hash_input = [str(swagger)]
         if openapi_version:
             hash_input.append(str(openapi_version))
+        if domain:
+            hash_input.append(self._X_HASH_DELIMITER)
```

Comment:
Really happy to see you're ensuring we trigger a new deployment on custom domain changes! But why are you adding an extra delimiter here? The join below will add the delimiter between all entries in the list.

Also, we need to be careful here with doing simple if checks for appending to the hash. What if we had 2 different parameters, x and y that are part of the hash with logic like this:

```python
    if x:
        hash_input.append(x)
    if y:
        hash_input.append(y)
```

Then I have a template where I set x to 'foo' and don't set y. Then later, I remove the x property and set y to 'foo'. The deployment hash will be the same even though I've made an actual change to my template. We should be building this array such that we avoid this problem. For example, we could avoid the above scenario by doing this:

```python
    if x:
        hash_input.append('x:{}'.format(x))
    if y:
        hash_input.append('y:{}'.format(y))
```

Whatever we do, we need to make it backwards-compatible with the current behavior, but moving forward, we should do something like this with new values that we decide to add to the hash array.

Reply:
Ah ok, then this case is probably satisfied and we should only apply my solution to strings or other properties that could be ambiguous.